### PR TITLE
Configurable time bound

### PR DESCRIPTION
**Pull request by geraldus**
@snoyberg , ok, it seems I've done. Please review, I can remove last two commits. Also I didn't bumped version, because I'm not sure is it breaking change or not.

### CODE REVIEW

File: Keter/Proxy.hs [hunk 1]
```diff
@@ -62,28 +62,25 @@ reverseProxy useHeader manager hostLookup listener =
...
                 else SIHFromSocket
-        } manager req0 sendResponse)
-  where
+        } manager req0 sendResponse
     protocol
         | isSecure = "https"
         | otherwise = "http"
 
-    -- FIXME This is a temporary workaround for
-    -- https://github.com/snoyberg/keter/issues/29. After some research, it
```
Comment (snoyberg) [inline Keter/Proxy.hs]: I'd still like a reference to that issue around the timeBound code, as it's likely to confuse someone in the future.

**Comment by snoyberg** [comment]
I just had one comment regarding leaving in a reference to issue #29. Otherwise, this looks great!

File: Keter/Proxy.hs (hunk 1 above)
Comment (geraldus) [inline Keter/Proxy.hs]: Well, I'll restore this right now (:

**Comment by geraldus** [comment]
Ready!

**Comment by snoyberg** [comment]
Thanks! Releasing to Hackage now.

File: Keter/Types/V04.hs [hunk 2]
```diff
@@ -92,6 +93,7 @@ instance Default KeterConfig where
         , kconfigSetuid = Nothing
         , kconfigReverseProxy = Set.empty
         , kconfigIpFromHeader = False
+        , kconfigConnectionTimeBound = 5000
```
Comment (tolysz) [inline Keter/Types/V04.hs]: This is 5s not 5 minutes... shhould be: 
5 * 60 * 1000

File: Keter/Types/V04.hs [hunk 3]
```diff
@@ -104,6 +106,7 @@ instance ParseYamlFile KeterConfig where
         <*> o .:? "setuid"
         <*> o .:? "reverse-proxy" .!= Set.empty
         <*> o .:? "ip-from-header" .!= False
+        <*> o .:? "connection-time-bound" .!= 5000
```
Comment (tolysz) [inline Keter/Types/V04.hs]: same: should be 5 minutes

File: Keter/Types/V10.hs [hunk 4]
```diff
@@ -134,6 +139,7 @@ instance Default KeterConfig where
         , kconfigExternalHttpPort = 80
         , kconfigExternalHttpsPort = 443
         , kconfigEnvironment = Map.empty
+        , kconfigConnectionTimeBound = 5000
```
Comment (tolysz) [inline Keter/Types/V10.hs]: same 5 minutes

File: Keter/Types/V10.hs [hunk 5]
```diff
@@ -153,6 +159,7 @@ instance ParseYamlFile KeterConfig where
             <*> o .:? "external-http-port" .!= 80
             <*> o .:? "external-https-port" .!= 443
             <*> o .:? "env" .!= Map.empty
+            <*> o .:? "connection-time-bound" .!= 5000
```
Comment (tolysz) [inline Keter/Types/V10.hs]: same 5 minutes

File: Keter/Types/V10.hs (hunk 4 above)
Comment (snoyberg) [inline Keter/Types/V10.hs]: Good catches, thank you. Pushing a commit to fix that now.

Comment (geraldus) [inline Keter/Types/V10.hs]: Ooops! Excuse me! Also `etc/keter-config.yaml` need to be updated (same value there).

Comment (snoyberg) [inline Keter/Types/V10.hs]: This should now be fixed.

**Pull request merged**